File: flux-flix-service/src/main/java/com/example/fluxflixservice/DataLoader.java
package com.example.fluxflixservice;

import org.springframework.stereotype.Component;
import reactor.core.publisher.Flux;

import javax.annotation.PostConstruct;

@Component
public class DataLoader {
    private final MovieRepository movieRepository;

    public DataLoader(MovieRepository movieRepository) {
        this.movieRepository = movieRepository;
    }

    @PostConstruct
    private void loadData() {
        this.movieRepository.deleteAll().thenMany(
                Flux.just("Enter the Mono<Void>",
                        "Y Tu Mono También",
                        "Meet the Fluxers")
                        .map(Movie::new)
                        .flatMap(this.movieRepository::save))
                .subscribe(System.out::println);

//        System.out.println("\nAll records in DB:\n");
//        this.movieRepository
//                .findAll()
//                .subscribe(movie -> System.out.println("> Found in DB: " + movie.toString()));
    }
}

File: flux-flix-service-kotlin/src/main/kotlin/com/example/fluxflixservicekotlin/DataLoader.kt
package com.example.fluxflixservicekotlin

import org.springframework.stereotype.Component
import reactor.core.publisher.Flux
import javax.annotation.PostConstruct

@Component
class DataLoader(private val movieRepository: MovieRepository) {
    @PostConstruct
    fun loadData() {
        movieRepository.deleteAll().thenMany(
                Flux.just("Enter the Mono<Void>",
                        "Y Tu Mono También",
                        "Meet the Fluxers")
                        .map { Movie(title = it) }
                        .flatMap { movieRepository.save(it) })
                .subscribe { println(it) }
    }
}